leakcanary-android-process/src/main/java/leakcanary/LeakCanaryProcess.kt

```diff
@@ -26,84 +21,30 @@ object LeakCanaryProcess {
    * a different process than the normal app process.
    */
   fun isInAnalyzerProcess(context: Context): Boolean {
-    val analyzerServiceClass: Class<out Service>
-    @Suppress("UNCHECKED_CAST")
-    try {
-      analyzerServiceClass =
-        Class.forName("leakcanary.internal.HeapAnalyzerService") as Class<out Service>
-    } catch (e: Exception) {
-      return false
-    }
-
     var isInAnalyzerProcess: Boolean? = isInAnalyzerProcess
     // This only needs to be computed once per process.
     if (isInAnalyzerProcess == null) {
-      isInAnalyzerProcess = isInServiceProcess(context, analyzerServiceClass)
+      isInAnalyzerProcess = isInWorkerProcess(context)
       this.isInAnalyzerProcess = isInAnalyzerProcess
     }
     return isInAnalyzerProcess
   }
 
-  @Suppress("ReturnCount")
-  private fun isInServiceProcess(
-    context: Context,
-    serviceClass: Class<out Service>
+  private fun isInWorkerProcess(
+    context: Context
   ): Boolean {
-    val packageManager = context.packageManager
-    val packageInfo: PackageInfo
-    try {
-      packageInfo = packageManager.getPackageInfo(context.packageName, PackageManager.GET_SERVICES)
-    } catch (e: Exception) {
-      SharkLog.d(e) { "Could not get package info for ${context.packageName}" }
-      return false
-    }
-
-    val mainProcess = packageInfo.applicationInfo.processName
+    var workerRunning = false
+    val workManager = WorkManager.getInstance(context)
+    val activeWorkers = workManager.getWorkInfosByTag("heap_analyzer_worker").get()
```

Comment:
one thing to be careful about here is that the process might be started even if no work is enqueued, for whatever reason.

Reply:
What do you suggest? Checking if either of the condition returns true?

Reply:
Most likely there won't be a change as we'll bind to a new service here anyway, I think, with RemoteListenableWorker